### PR TITLE
Added DELIMITER support to the dolt shell

### PR DESCRIPTION
**Pull request by Hydrocharged**

### CODE REVIEW

**Comment by Hydrocharged** [comment]
This will fail all of the tests until I can make a PR against https://github.com/dolthub/ishell, then I'll update this one with the proper commit.

File: bats/sql-delimiter.expect [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/expect
```
Comment (zachmu) [inline bats/sql-delimiter.expect:1]: Probably want this to include at least one test of multiple statements like a `BEGIN .. END` block. The only place this is supported in the grammar right now is when creating a trigger.

File: go/cmd/dolt/commands/sql.go [hunk 2]
```diff
@@ -638,27 +638,53 @@ func runShell(ctx *sql.Context, se *sqlEngine, mrEnv env.MultiRepoEnv, initialRo
...
-			verr := formatQueryError("", err)
-			shell.Println(verr.Verbose())
-		} else if rowIter != nil {
-			err = PrettyPrintResults(ctx, se.resultFormat, sqlSch, rowIter)
-			if err != nil {
-				shell.Println(color.RedString(err.Error()))
+		shouldProcessQuery := true
+
+		//TODO: Parse this in a way that can handle comments
+		if len(singleLine) > 9 && strings.HasPrefix(strings.ToLower(singleLine[:9]), "delimiter") {
```
Comment (zachmu) [inline go/cmd/dolt/commands/sql.go:649]: Probably feel better about a regex match than doing prefixes and string splits

File: go/go.mod [hunk 3]
```diff
@@ -78,4 +78,6 @@ require (
 
 replace github.com/dolthub/dolt/go/gen/proto/dolt/services/eventsapi => ./gen/proto/dolt/services/eventsapi
 
+replace github.com/dolthub/ishell => ../../ishell
```
Comment (zachmu) [inline go/go.mod:81]: Don't check this in

**Review by zachmu — APPROVED** [review]
LG, just a couple comments

**Pull request merged**